cmd/master-controller-manager/main.go

```diff
@@ -131,6 +131,11 @@ func main() {
 			log.Fatalw("invalid KubermaticConfiguration", zap.Error(err))
 		}
 	}
+	if config != nil {
+		versions := kubermatic.NewDefaultVersions()
+		config.Status.KubermaticEdition = versions.KubermaticCommit
+		config.Status.KubermaticVersion = versions.KubermaticEdition.String()
+	}
```

Comment:
This will have absolutely no effect. If your plan was to update the KubermaticConfiguration status, you must do so with a Kubernetes client and not just this variable. `config` here is only set when running in dev mode and loading the config from a file. In real world scenarios, the config can and will change _at runtime_,. Also this way you only change it for the master-ctrl-mgr, what about the operator, seed-ctrl-mgr and oother components?

You need to write a controller that watches the KubermaticConfiguration and patches the status to update the version.

Reply:
@xrstf thanks!
sorry, should have put `WIP` in the PR.
these particular changes were made to confirm if the value sources are correct or not?
i.e versions.KubermaticCommit for version and versions.KubermaticEdition for edition.
I will work on the Controller design, let me know if the sources are not correct? :)

Reply:
config.Status.KubermaticEdition = versions.KubermaticCommit
config.Status.KubermaticVersion = versions.KubermaticEdition.String()

You mixed them up, I think ;-)

Reply:
Ya sorry meant this, config.Status.KubermaticEdition = versions.KubermaticEdition.String()
config.Status.KubermaticVersion = versions.KubermaticCommit